client/src/lib/components/DatasetForm/_ExtraField.svelte

```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+  import type { ExtraField } from "src/definitions/catalogs";
+  import { toSelectOptions } from "src/lib/transformers/form";
+  import { renderMarkdown } from "src/lib/util/markdown";
+  import InputField from "../InputField/InputField.svelte";
+  import RadioGroupField from "../RadioGroupField/RadioGroupField.svelte";
+  import Select from "../Select/Select.svelte";
+
+  export let extraField: ExtraField;
+  export let value = "";
+
+  // NOTE: hint texts of extra fields are considered TRUSTED as catalogs can
+  // only be created via an approved PR on https://github.com/etalab/catalogage-donnees-config.
+  $: hintTextHtml = {
+    isHTML: true,
+    content: renderMarkdown(extraField.hintText),
+  };
+
+  const makeBoolOptions = (trueValue: string, falseValue: string) => {
+    return toSelectOptions({
+      [trueValue]: trueValue,
+      [falseValue]: falseValue,
+    });
+  };
+
+  const makeEnumOptions = (values: string[]) => {
+    const map = {};
+    values.forEach((value) => (map[value] = value));
+    return toSelectOptions(map);
+  };
+</script>
+
+{#if extraField.type === "TEXT"}
```

Comment:
Est-ce que ça ne vaudrait pas le coup de créer une constants pour le type d'extrafield ?

Reply:
Pas vraiment, car `type` est ici une [discriminated union](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions).

Si les valeurs possibles de `.type` changent de sorte que `"TEXT"` ne soit plus possible, on obtiendrait la même erreur que dans la doc ci-dessus:

> This condition will always return 'false' since the types '"BOOL" | "ENUM"' and '"TEXT"' have no overlap.ts(2367)